In [139]:
# IMPORT NECESSARY LIBRARIES

import pandas as pd
import numpy as np
from fuzzywuzzy import process, fuzz

In [140]:
# OPEN THE FILE AND STORE IN OUR DATAFRAME

audits_15_20 = pd.read_csv('/Users/leona/Documents/Python/Contraloria/resol_publics_audits_15-20.csv', index_col=False)
audits_15_20

Unnamed: 0    institucion                                file_name  \
0              0       CONACAFE            RIA-CGR-001-15-AFYC-CONACAFE    
1              1  CiudadSandino    RIA-CGR-002-15-AEES-AM-CiudadSandino    
2              2     RubenDario       RIA-CGR-003-15-AEES-TN-RubenDario    
3              3        Acoyapa          RIA-CGR-004-15-AEES-AM-Acoyapa    
4              4      ElCrucero        RIA-CGR-005-15-AEES-AM-ElCrucero    
...          ...            ...                                      ...   
5814        5814      ElCrucero      RIA-UAI-927-2020-ACUM-AM-ElCrucero    
5815        5815  CiudadSandino  RIA-UAI-928-2020-ACUM-AM-CiudadSandino    
5816        5816  CiudadSandino  RIA-UAI-929-2020-ACUM-AM-CiudadSandino    
5817        5817      Matagalpa      RIA-UAI-930-2020-ACUM-AM-Matagalpa    
5818        5818        Corinto        RIA-UAI-931-2020-ACUM-AM-Corinto    

                                                    url   año  \
0     /index.php/repositorio/send/23-cgr/1977-ria-cg...  2015   
1     /index.php/repositorio/send/23-cgr/1935-ria-cg...  2015   
2     /index.php/repositorio/send/23-cgr/5014-ria-cg...  2015   
3     /index.php/repositorio/send/23-cgr/1919-ria-cg...  2015   
4     /index.php/repositorio/send/23-cgr/1926-ria-cg...  2015   
...                                                 ...   ...   
5814  /index.php/repositorio/send/113-uai/11866-ria-...  2020   
5815  /index.php/repositorio/send/113-uai/11865-ria-...  2020   
5816  /index.php/repositorio/send/113-uai/11864-ria-...  2020   
5817  /index.php/repositorio/send/113-uai/11863-ria-...  2020   
5818  /index.php/repositorio/send/113-uai/11862-ria-...  2020   

        Unidad_auditora  
0        Fondos Propios  
1        Fondos Propios  
2        Fondos Propios  
3        Fondos Propios  
4        Fondos Propios  
...                 ...  
5814  Auditoría interna  
5815  Auditoría interna  
5816  Auditoría interna  
5817  Auditoría interna  
5818  Auditoría interna  

[5819 rows x 6 columns]

## As i said, we're going to need to get a list with the values/names more accurate as possible. We have that information in that data set but it's a lot.

**The  first thing to do is get a dataframe to know the match_score for every name. In this case it's going to get a dataframe with 3 column: the original column, the match and the score. The more close to 100 the more similar is. So, with process.extract it is created a tuple with this information, which is going to be converted as DataFrame**

In [141]:
score_sort = [(x,) + i
             for x in audits_15_20['institucion'].values 
             for i in process.extract(x, audits_15_20['institucion'].values , scorer=fuzz.token_set_ratio)]

In [142]:
#Create a dataframe from the tuples

similarity_sort = pd.DataFrame(score_sort, columns=['inst_sort','match_sort','score_sort'])
similarity_sort.sort_values(by=['score_sort'])

inst_sort                        match_sort  \
3869   BenemeritoCuerpoBomberosManagua  UnidadEjecutoraProgramaUsuraCero   
3868   BenemeritoCuerpoBomberosManagua             ErnestoSequeiraBlanco   
3867   BenemeritoCuerpoBomberosManagua         CentroNacionalNoraAstorga   
3504                          Mulukuku                            MuyMuy   
3744                          Mulukuku                            MuyMuy   
...                                ...                               ...   
10148                             INSS                              INSS   
10147                             INSS                              INSS   
10146                             INSS                              INSS   
10157                             INSS                              INSS   
29094                          Corinto                           Corinto   

       score_sort  
3869           41  
3868           42  
3867           43  
3504           43  
3744           43  
...           ...  
10148         100  
10147         100  
10146         100  
10157         100  
29094         100  

[29095 rows x 3 columns]

In [143]:
# WE SELECT ONLY THE DATA WHERE SCORE_SORT IS EQUAL TO 100

higher_match = similarity_sort[similarity_sort['score_sort'] == 100 ]
higher_match

inst_sort     match_sort  score_sort
0           CONACAFE       CONACAFE         100
1           CONACAFE       CONACAFE         100
2           CONACAFE       CONACAFE         100
3           CONACAFE       CONACAFE         100
5      CiudadSandino  CiudadSandino         100
...              ...            ...         ...
29090        Corinto        Corinto         100
29091        Corinto        Corinto         100
29092        Corinto        Corinto         100
29093        Corinto        Corinto         100
29094        Corinto        Corinto         100

[28187 rows x 3 columns]

**Now, i will check for the names in the match_sort columns to know the more accurate names of the "institucion" column**

In [144]:
higher_match_list = higher_match.match_sort.sort_values().unique().tolist()
print(higher_match_list)

['ALMA', 'AN', 'ANA', 'ASAMBLEA', 'Achuapa', 'Acoyapa', 'AldoChavarria', 'Aleman', 'AlemanNicaraguense', 'Altagracia', 'Asamblea', 'AsambleaNacional', 'BCN', 'BFP', 'BICU', 'BancoProduzcamos', 'Belen', 'BenemeritoCuerpoBomberosManagua', 'BerthaCalderon', 'Bluefields', 'Boaco', 'BocanadePaiwas', 'Bonanza', 'BuenosAires', 'CAM', 'CENAO', 'CETREX', 'CGR', 'CINEMATECA', 'CNM', 'CNPC', 'CNU', 'CNZF', 'COERCO', 'CONACAFE', 'CONAMI', 'CONCAFE', 'CORNAP', 'CORREOS', 'CRAAN', 'CRACCN', 'CRACCS', 'CRCCN', 'CRRACCN', 'CSE', 'CSJ', 'CZF', 'Camoapa', 'Carazo', 'Cardenas', 'Catarina', 'CentroNacionalNoraAstorga', 'CentroNacionaldeCardiologia', 'Chichigalpa', 'Chinandega', 'CincoPinos', 'CiudadAntigua', 'CiudadDario', 'CiudadSandino', 'Cofradia', 'Comalapa', 'Condega', 'Corinto', 'CornIsland', 'CorreosdeNicaragua', 'CorreosdeNicarauga', 'CorresodeNicaragua', 'Cusmapa', 'DG I', 'DGA', 'DGI', 'DGME', 'DGSA', 'DGSPN', 'DesembocaduradeRioGrande', 'Dipilto', 'Diria', 'Diriamba', 'Diriomo', 'DoctorRobertoC

In [145]:
print("The number of Institutucion now is:",higher_match.match_sort.nunique() )

The number of Institutucion now is: 359


### After running the code above, we can check that there are still some similar names like "asamblea" and "AsambleaNacional". The number of "institucion" reduced to 359 from 381

### The next step is to add a new column with the best matchs for institutions. To do that, the below code is necessary.

In [146]:
# Create a function in order to store the new values

institucion_bien=[]
proporcion = []

def cambios(mal, bien):
    for i in mal:
        x=process.extractOne(i, bien)
        institucion_bien.append(x[0])
        proporcion.append(x[1])
    return institucion_bien, proporcion

mal = audits_15_20['institucion'].values
bien = higher_match_list


corregido, rango = cambios(mal, bien)

In [147]:
# ADD NEW COLUMNS WITH NEW VALUES "inst_buenas" and "ratiom", in order to compare how was matched the
# "institucion" column

audits_15_20['inst_buenas'] = pd.Series(corregido)
audits_15_20['ratiom'] = pd.Series(rango)

audits_15_20

Unnamed: 0    institucion                                file_name  \
0              0       CONACAFE            RIA-CGR-001-15-AFYC-CONACAFE    
1              1  CiudadSandino    RIA-CGR-002-15-AEES-AM-CiudadSandino    
2              2     RubenDario       RIA-CGR-003-15-AEES-TN-RubenDario    
3              3        Acoyapa          RIA-CGR-004-15-AEES-AM-Acoyapa    
4              4      ElCrucero        RIA-CGR-005-15-AEES-AM-ElCrucero    
...          ...            ...                                      ...   
5814        5814      ElCrucero      RIA-UAI-927-2020-ACUM-AM-ElCrucero    
5815        5815  CiudadSandino  RIA-UAI-928-2020-ACUM-AM-CiudadSandino    
5816        5816  CiudadSandino  RIA-UAI-929-2020-ACUM-AM-CiudadSandino    
5817        5817      Matagalpa      RIA-UAI-930-2020-ACUM-AM-Matagalpa    
5818        5818        Corinto        RIA-UAI-931-2020-ACUM-AM-Corinto    

                                                    url   año  \
0     /index.php/repositorio/send/23-cgr/1977-ria-cg...  2015   
1     /index.php/repositorio/send/23-cgr/1935-ria-cg...  2015   
2     /index.php/repositorio/send/23-cgr/5014-ria-cg...  2015   
3     /index.php/repositorio/send/23-cgr/1919-ria-cg...  2015   
4     /index.php/repositorio/send/23-cgr/1926-ria-cg...  2015   
...                                                 ...   ...   
5814  /index.php/repositorio/send/113-uai/11866-ria-...  2020   
5815  /index.php/repositorio/send/113-uai/11865-ria-...  2020   
5816  /index.php/repositorio/send/113-uai/11864-ria-...  2020   
5817  /index.php/repositorio/send/113-uai/11863-ria-...  2020   
5818  /index.php/repositorio/send/113-uai/11862-ria-...  2020   

        Unidad_auditora    inst_buenas  ratiom  
0        Fondos Propios       CONACAFE     100  
1        Fondos Propios  CiudadSandino     100  
2        Fondos Propios     RubenDario     100  
3        Fondos Propios        Acoyapa     100  
4        Fondos Propios      ElCrucero     100  
...                 ...            ...     ...  
5814  Auditoría interna      ElCrucero     100  
5815  Auditoría interna  CiudadSandino     100  
5816  Auditoría interna  CiudadSandino     100  
5817  Auditoría interna      Matagalpa     100  
5818  Auditoría interna        Corinto     100  

[5819 rows x 8 columns]

In [148]:
# TO CHECK THE UNIQUE VALUES IN THE NEW COLUMN WE RUN THE CODE

print(audits_15_20['inst_buenas'].sort_values().unique().tolist())

['ALMA', 'AN', 'ANA', 'ASAMBLEA', 'Achuapa', 'Acoyapa', 'AldoChavarria', 'Aleman', 'AlemanNicaraguense', 'Altagracia', 'AsambleaNacional', 'BCN', 'BFP', 'BICU', 'BancoProduzcamos', 'Belen', 'BenemeritoCuerpoBomberosManagua', 'BerthaCalderon', 'Bluefields', 'Boaco', 'BocanadePaiwas', 'Bonanza', 'BuenosAires', 'CAM', 'CENAO', 'CETREX', 'CGR', 'CINEMATECA', 'CNM', 'CNPC', 'CNU', 'CNZF', 'COERCO', 'CONACAFE', 'CONAMI', 'CONCAFE', 'CORNAP', 'CORREOS', 'CRAAN', 'CRACCN', 'CRACCS', 'CRCCN', 'CRRACCN', 'CSE', 'CSJ', 'CZF', 'Camoapa', 'Carazo', 'Cardenas', 'Catarina', 'CentroNacionalNoraAstorga', 'CentroNacionaldeCardiologia', 'Chichigalpa', 'Chinandega', 'CincoPinos', 'CiudadAntigua', 'CiudadDario', 'CiudadSandino', 'Cofradia', 'Comalapa', 'Condega', 'Corinto', 'CornIsland', 'CorreosdeNicaragua', 'CorreosdeNicarauga', 'CorresodeNicaragua', 'Cusmapa', 'DG I', 'DGA', 'DGI', 'DGME', 'DGSA', 'DGSPN', 'DesembocaduradeRioGrande', 'Dipilto', 'Diria', 'Diriamba', 'Diriomo', 'DoctorRobertoCalderon', 'D

**It is confirmed that it is necessary to fix again the names.**

In [149]:
# With this, it can be checked the columns that are differents between "institucion" and "inst_buenas"

audits_15_20[~audits_15_20['institucion'].isin(audits_15_20['inst_buenas'])]

Unnamed: 0        institucion  \
258          258         Villanueva   
310          310      SanJuandelSur   
315          315              Elcua   
581          581      SanJuandelSur   
709          709           Asamblea   
801          801      SanJuandelSur   
859          859              BELEN   
883          883      SanJuandelSur   
1495        1495      SanJuandelSur   
1549        1549      SanJuandelSur   
2609        2609      SanJuandelSur   
2629        2629      SanJuandelSur   
2852        2852         AM Nindiri   
3023        3023          .Jinotepe   
3579        3579          UsuraCero   
3637        3637    SanRafaelDelSur   
3638        3638    SanRafaelDelSur   
3739        3739      ElTumalaDalia   
3790        3790  SanRafaelDelNorte   
3839        3839      SanJuandelSur   
4056        4056       ProNicaragua   
4114        4114       ProNicaragua   
4213        4213        Presidencia   
4214        4214        Presidencia   
4267        4267        Presidencia   
4268        4268        Presidencia   
4392        4392        Presidencia   
4393        4393        Presidencia   
4429        4429            GRANADA   
4430        4430             ESTELI   
4821        4821    VicePresidencia   
4824        4824    VicePresidencia   
4886        4886          UsuraCero   
4930        4930          UsuraCero   
4955        4955          UsuraCero   
5065        5065       ProNicaragua   
5486        5486             SOMOTO   
5487        5487             OCOTAL   
5552        5552           Asamblea   
5663        5663        Presidencia   
5706        5706        Presidencia   

                                      file_name  \
258         RIA-CGR-1237-16-AFYC-AM-Villanueva    
310      RIA-CGR-1390-16-AEES-AM-SanJuandelSur    
315               RIA-CGR-538-16-AFYC-AM-Elcua    
581      RIA-CGR-1023-18-AFYC-AM-SanJuandelSur    
709              RIA-CGR-1147-19-AFYC-Asamblea    
801      RIA-CGR-2075-19-AFYC-AM-SanJuandelSur    
859            RIA-CGR-1575-2020-AFYC-AM-BELEN    
883    RIA-CGR-1743-2020-AFYC-AM-SanJuandelSur    
1495     RIA-UAI-1454-15-AEES-AM-SanJuandelSur    
1549      RIA-UAI-156-15-AEES-AM-SanJuandelSur    
2609     RIA-UAI-1219-16-AEES-AM-SanJuandelSur    
2629          RIA-UAI-1248-16-AM-SanJuandelSur    
2852            RIA-UAI-254-16-AEES-AM Nindiri    
3023          RIA-UAI-427-16-AEES-AM-.Jinotepe    
3579    RIA-UAI-080-17-AEES-Programa-UsuraCero    
3637   RIA-UAI-1090-17-AEES-AM-SanRafaelDelSur    
3638   RIA-UAI-1091-17-AEES-AM-SanRafaelDelSur    
3739      RIA-UAI-130-17-AEES-AM-ElTumalaDalia    
3790  RIA-UAI-213-17-AEES-AM-SanRafaelDelNorte    
3839      RIA-UAI-296-17-AEES-AM-SanJuandelSur    
4056          RIA-UAI-077-18-AEES-ProNicaragua    
4114          RIA-UAI-125-18-AEES-ProNicaragua    
4213           RIA-UAI-218-18-AEES-Presidencia    
4214           RIA-UAI-219-18-AEES-Presidencia    
4267           RIA-UAI-277-18-AEES-Presidencia    
4268           RIA-UAI-278-18-AEES-Presidencia    
4392           RIA-UAI-412-18-AEES-Presidencia    
4393           RIA-UAI-413-18-AEES-Presidencia    
4429            RIA-UAI-459-18-AEES-AM-GRANADA    
4430             RIA-UAI-460-18-AEES-AM-ESTELI    
4821      RIA-UAI-1932-19-ACUM-VicePresidencia    
4824      RIA-UAI-1935-19-ACUM-VicePresidencia    
4886             RIA-UAI-214-19-ACUM-UsuraCero    
4930             RIA-UAI-246-19-ACUM-UsuraCero    
4955             RIA-UAI-272-19-AFYC-UsuraCero    
5065          RIA-UAI-420-19-ACUM-ProNicaragua    
5486          RIA-UAI-1704-2020-ACUM-AM-SOMOTO    
5487          RIA-UAI-1705-2020-ACUM-AM-OCOTAL    
5552           RIA-UAI-1886-2020-ACUM-Asamblea    
5663         RIA-UAI-310-2020-ACUM-Presidencia    
5706         RIA-UAI-483-2020-ACUM-Presidencia    

                                                    url   año  \
258   /index.php/repositorio/send/20-cgr/4767-ria-cg...  2016   
310   /index.php/repositorio/send/20-cgr/5756-ria-cg...  2016   
315   /index.php/repositorio/

**Now, the names will be replaced manually to get the most accurate names**

In [150]:
audits_15_20['inst_buenas'] = audits_15_20['inst_buenas'].replace({'ASAMBLEA':'AsambleaNacional',
        'GRAAS':'GRACCS','GRAAN':'GRACCN',  'CorreosdeNicarauga':'CorreosdeNicaragua',
        'CorresodeNicaragua':'CorreosdeNicaragua',
         'Kukarahill':'KukraHill','LN':'LoteriaNacional','LOTERIA':'LoteriaNacional',
         'LoterianNacional':'LoteriaNacional','LaCruzRioGrande':'LaCruzdeRioGrande',
         'LaPaz':'LaPazCarazo','MAG':'MAGFOR','MACFOR':'MAGFOR',
         'MEDE':'MINED','MEFFCA':'MEFCCA','MIFAMILIA':'MIFAN',
         'MINGOB':'MIGOB','MINISTERIOMUJER':'MINIM','MATIGUA':'MATIGUAS',
         'MinisterioPublico':'MP','MinisteriodelaMujer':'MINIM',
         'MinisteriodelaJuventud':'MINJUVE','Mnagua':'Managua','NSS':'INSS',
         'NuevoAmanecerNancyBach':'HospitalNuevoAmanecer','PPDDH':'PPDH',
         'PresidenciaRepublica':'PRESIDENCIA','PresidenciadelaRepublica':'PRESIDENCIA',
         'Prinzapolca':'Prinzapolka','RegionalDoctorErnestoSequeira':'HospitalRegionalDoctorErnestoSequeira',
         'SanJuanSur':'SanJuanDelSur','SanRaafeldelSur':'SanRafaeldelSur',
         'SucursalChinandega':'INISER_SucursalChinandega', 'CONCAFE':'CONACAFE',
         'UnidadEjecutoraProgramaUsuraCero':'USURACERO','VIillaNueva':'VillaNueva','Wasalala':'Waslala',
        'CRAAN':'CRACCN','CRCCN':'CRACCN','GRACS':'GRACCS', 'AlemanNicaraguense':'HospitalAlemanNicaraguense',
        'AldoChavarria':'HospitalAldoChavarria','Aleman':'HospitalAlemanNicaraguense','DG I':'DGI',
        'AN':'AsambleaNacional'})
                                                                   

In [151]:
print(audits_15_20['inst_buenas'].sort_values().unique().tolist())

['ALMA', 'ANA', 'Achuapa', 'Acoyapa', 'Altagracia', 'AsambleaNacional', 'BCN', 'BFP', 'BICU', 'BancoProduzcamos', 'Belen', 'BenemeritoCuerpoBomberosManagua', 'BerthaCalderon', 'Bluefields', 'Boaco', 'BocanadePaiwas', 'Bonanza', 'BuenosAires', 'CAM', 'CENAO', 'CETREX', 'CGR', 'CINEMATECA', 'CNM', 'CNPC', 'CNU', 'CNZF', 'COERCO', 'CONACAFE', 'CONAMI', 'CORNAP', 'CORREOS', 'CRACCN', 'CRACCS', 'CRRACCN', 'CSE', 'CSJ', 'CZF', 'Camoapa', 'Carazo', 'Cardenas', 'Catarina', 'CentroNacionalNoraAstorga', 'CentroNacionaldeCardiologia', 'Chichigalpa', 'Chinandega', 'CincoPinos', 'CiudadAntigua', 'CiudadDario', 'CiudadSandino', 'Cofradia', 'Comalapa', 'Condega', 'Corinto', 'CornIsland', 'CorreosdeNicaragua', 'Cusmapa', 'DGA', 'DGI', 'DGME', 'DGSA', 'DGSPN', 'DesembocaduradeRioGrande', 'Dipilto', 'Diria', 'Diriamba', 'Diriomo', 'DoctorRobertoCalderon', 'Dolores', 'EAAI', 'ECONS3', 'EMCOSE', 'ENABAS', 'ENACAL', 'ENATREL', 'ENEL', 'ENIMEX', 'ENIMPORT', 'EPN', 'ElAlmendro', 'ElAyote', 'ElCastillo', 'ElC

### According with above information, it remains some problems with the names of "institucion", so i will correct them manually again. 

### Also, i will check for a particular name "UNAN"because it can be UNANLeon or UNANManagua. In order to do that, i need to check in the downloaded files

In [152]:
audits_15_20['inst_buenas'] = audits_15_20['inst_buenas'].replace({'ElTuma':'ElTumaLaDalia',
                            'ErnestoSequeiraBlanco':'HospitalRegionalDoctorErnestoSequeira',
                            'IES':'IESUNI','INISER_SucursalChinandega':'INISER','MCHP':'MHCP',
                            'MHP':'MHCP','MIFAN':'MIFAM','Mateares':'Mateare','Matigua':'Matiguas',
                            'Ticantepe':'Ticuantepe'})

In [153]:
audits_15_20[audits_15_20['inst_buenas'] == 'UNAN']

Unnamed: 0 institucion                     file_name  \
3292        3292        UNAN     RIA-UAI-753-16-AEES-UNAN    
3609        3609        UNAN  RIA-UAI-1019-2017-AEES-UNAN    
3923        3923        UNAN     RIA-UAI-844-17-AEES-UNAN    

                                                    url   año  \
3292  /index.php/repositorio/send/21-uai/417-ria-uai...  2016   
3609  /index.php/repositorio/send/63-uai/7189-ria-ua...  2017   
3923  /index.php/repositorio/send/63-uai/6560-ria-ua...  2017   

        Unidad_auditora inst_buenas  ratiom  
3292  Auditoría interna        UNAN     100  
3609  Auditoría interna        UNAN     100  
3923  Auditoría interna        UNAN     100

In [154]:
audits_15_20.loc[audits_15_20.file_name == 'RIA-UAI-1019-2017-AEES-UNAN', 'inst_buenas'] = "UNANManagua"

In [155]:
audits_15_20[audits_15_20['inst_buenas'] == 'UNAN']

Unnamed: 0 institucion                     file_name  \
3292        3292        UNAN     RIA-UAI-753-16-AEES-UNAN    
3609        3609        UNAN  RIA-UAI-1019-2017-AEES-UNAN    
3923        3923        UNAN     RIA-UAI-844-17-AEES-UNAN    

                                                    url   año  \
3292  /index.php/repositorio/send/21-uai/417-ria-uai...  2016   
3609  /index.php/repositorio/send/63-uai/7189-ria-ua...  2017   
3923  /index.php/repositorio/send/63-uai/6560-ria-ua...  2017   

        Unidad_auditora inst_buenas  ratiom  
3292  Auditoría interna        UNAN     100  
3609  Auditoría interna        UNAN     100  
3923  Auditoría interna        UNAN     100

In [156]:
audits_15_20.loc[audits_15_20.file_name == 'RIA-UAI-753-16-AEES-UNAN', 'inst_buenas'] = "UNANLeon"

In [157]:
audits_15_20[audits_15_20['inst_buenas'] == 'UNAN']

Unnamed: 0 institucion                     file_name  \
3292        3292        UNAN     RIA-UAI-753-16-AEES-UNAN    
3609        3609        UNAN  RIA-UAI-1019-2017-AEES-UNAN    
3923        3923        UNAN     RIA-UAI-844-17-AEES-UNAN    

                                                    url   año  \
3292  /index.php/repositorio/send/21-uai/417-ria-uai...  2016   
3609  /index.php/repositorio/send/63-uai/7189-ria-ua...  2017   
3923  /index.php/repositorio/send/63-uai/6560-ria-ua...  2017   

        Unidad_auditora inst_buenas  ratiom  
3292  Auditoría interna        UNAN     100  
3609  Auditoría interna        UNAN     100  
3923  Auditoría interna        UNAN     100

In [158]:
audits_15_20.loc[audits_15_20.file_name == 'RIA-UAI-844-17-AEES-UNAN', 'inst_buenas'] = "UNANLeon"

In [159]:
audits_15_20[audits_15_20['inst_buenas'] == 'UNAN']

Unnamed: 0 institucion                     file_name  \
3292        3292        UNAN     RIA-UAI-753-16-AEES-UNAN    
3609        3609        UNAN  RIA-UAI-1019-2017-AEES-UNAN    
3923        3923        UNAN     RIA-UAI-844-17-AEES-UNAN    

                                                    url   año  \
3292  /index.php/repositorio/send/21-uai/417-ria-uai...  2016   
3609  /index.php/repositorio/send/63-uai/7189-ria-ua...  2017   
3923  /index.php/repositorio/send/63-uai/6560-ria-ua...  2017   

        Unidad_auditora inst_buenas  ratiom  
3292  Auditoría interna        UNAN     100  
3609  Auditoría interna        UNAN     100  
3923  Auditoría interna        UNAN     100

In [160]:
# NOW, WE CHECK AGAIN FOR THE UNIQUE VALUES IN "inst_buenas"

print(audits_15_20['inst_buenas'].sort_values().unique().tolist())

['ALMA', 'ANA', 'Achuapa', 'Acoyapa', 'Altagracia', 'AsambleaNacional', 'BCN', 'BFP', 'BICU', 'BancoProduzcamos', 'Belen', 'BenemeritoCuerpoBomberosManagua', 'BerthaCalderon', 'Bluefields', 'Boaco', 'BocanadePaiwas', 'Bonanza', 'BuenosAires', 'CAM', 'CENAO', 'CETREX', 'CGR', 'CINEMATECA', 'CNM', 'CNPC', 'CNU', 'CNZF', 'COERCO', 'CONACAFE', 'CONAMI', 'CORNAP', 'CORREOS', 'CRACCN', 'CRACCS', 'CRRACCN', 'CSE', 'CSJ', 'CZF', 'Camoapa', 'Carazo', 'Cardenas', 'Catarina', 'CentroNacionalNoraAstorga', 'CentroNacionaldeCardiologia', 'Chichigalpa', 'Chinandega', 'CincoPinos', 'CiudadAntigua', 'CiudadDario', 'CiudadSandino', 'Cofradia', 'Comalapa', 'Condega', 'Corinto', 'CornIsland', 'CorreosdeNicaragua', 'Cusmapa', 'DGA', 'DGI', 'DGME', 'DGSA', 'DGSPN', 'DesembocaduradeRioGrande', 'Dipilto', 'Diria', 'Diriamba', 'Diriomo', 'DoctorRobertoCalderon', 'Dolores', 'EAAI', 'ECONS3', 'EMCOSE', 'ENABAS', 'ENACAL', 'ENATREL', 'ENEL', 'ENIMEX', 'ENIMPORT', 'EPN', 'ElAlmendro', 'ElAyote', 'ElCastillo', 'ElC

In [161]:
print(audits_15_20['institucion'].nunique())
print(audits_15_20['inst_buenas'].nunique())

370
302


## As shown below, we can see that we reduced the number of names to 301, and we can see that there are no more duplicates/similar values

In [164]:
audits_15_20.drop('ratiom', axis=1, inplace=True)

In [ ]:
audits_15_20.drop('Unnamed: 0', axis=1, inplace=True)

In [166]:
audits_15_20

institucion                                file_name  \
0          CONACAFE            RIA-CGR-001-15-AFYC-CONACAFE    
1     CiudadSandino    RIA-CGR-002-15-AEES-AM-CiudadSandino    
2        RubenDario       RIA-CGR-003-15-AEES-TN-RubenDario    
3           Acoyapa          RIA-CGR-004-15-AEES-AM-Acoyapa    
4         ElCrucero        RIA-CGR-005-15-AEES-AM-ElCrucero    
...             ...                                      ...   
5814      ElCrucero      RIA-UAI-927-2020-ACUM-AM-ElCrucero    
5815  CiudadSandino  RIA-UAI-928-2020-ACUM-AM-CiudadSandino    
5816  CiudadSandino  RIA-UAI-929-2020-ACUM-AM-CiudadSandino    
5817      Matagalpa      RIA-UAI-930-2020-ACUM-AM-Matagalpa    
5818        Corinto        RIA-UAI-931-2020-ACUM-AM-Corinto    

                                                    url   año  \
0     /index.php/repositorio/send/23-cgr/1977-ria-cg...  2015   
1     /index.php/repositorio/send/23-cgr/1935-ria-cg...  2015   
2     /index.php/repositorio/send/23-cgr/5014-ria-cg...  2015   
3     /index.php/repositorio/send/23-cgr/1919-ria-cg...  2015   
4     /index.php/repositorio/send/23-cgr/1926-ria-cg...  2015   
...                                                 ...   ...   
5814  /index.php/repositorio/send/113-uai/11866-ria-...  2020   
5815  /index.php/repositorio/send/113-uai/11865-ria-...  2020   
5816  /index.php/repositorio/send/113-uai/11864-ria-...  2020   
5817  /index.php/repositorio/send/113-uai/11863-ria-...  2020   
5818  /index.php/repositorio/send/113-uai/11862-ria-...  2020   

        Unidad_auditora    inst_buenas  
0        Fondos Propios       CONACAFE  
1        Fondos Propios  CiudadSandino  
2        Fondos Propios     RubenDario  
3        Fondos Propios        Acoyapa  
4        Fondos Propios      ElCrucero  
...                 ...            ...  
5814  Auditoría interna      ElCrucero  
5815  Auditoría interna  CiudadSandino  
5816  Auditoría interna  CiudadSandino  
5817  Auditoría interna      Matagalpa  
5818  Auditoría interna        Corinto  

[5819 rows x 6 columns]

In [167]:
# SAVE THE DATASET

audits_15_20.to_csv('audits_publi_15_20.csv')